src/persist-client/src/internal/apply.rs

```diff
@@ -210,6 +211,28 @@ where
             })
     }
 
+    /// Returns the requested schema, if known at the current state.
+    ///
+    /// If None, returns the latest schema registered at the current state.
```

Comment:
This feels like an unusual semantics! Is this just a convenience for the migration? (If not, I'd be tempted to split this into separate `get` and `latest` methods for the some/none cases.

Reply:
I ran into something similar when working on the adapter changes and ended up making an enum like:

```
enum SchemaVersion {
  Latest,
  Specific(usize),
}
```

Then callers would use `SchemaVersion::Latest` instead of `None` which was quite nice. 

Not a blocker but using a specific enum or splitting this up into two methods would be nice